crypto/hmac/hmac.c

```diff
@@ -35,6 +35,10 @@ int HMAC_Init_ex(HMAC_CTX *ctx, const void *key, int len,
         return 0;
     }
 
+    /* We currently don't support shake128 and shake256 with HMAC */
+    if ((EVP_MD_meth_get_flags(md) & EVP_MD_FLAG_XOF) != 0)
+        return 0;
```

Comment:
I think -2 is the correct error code.

Reply:
Never mind, not for this function.

Reply:
One question for curiosity: why does none of the functions in crypto/hmac (and neither in crypto/sha) push an error message?